qiskit_ibm_runtime/ibm_backend.py

```diff
@@ -752,13 +753,13 @@ def _runtime_run(
         if self._service._channel == "ibm_quantum":
             hgp_name = self._instance or self._service._get_hgp().name
 
-        # Check if initialized within a Primitive session. If so, issue a warning.
-        if get_cm_primitive_session():
-            warnings.warn(
-                "A Primitive session is open but Backend.run() jobs will not be run within this session"
-            )
+        if session:
+            self._session = session
+        elif get_cm_session():
+            self._session = get_cm_session()
```

Comment:
```python
backend = self.service.get_backend("ibmq_qasm_simulator")
with Session(backend=backend) as session:
    sampler = Sampler(session=session)
    job1 = sampler.run(circuits=ReferenceCircuits.bell())
    job2 = backend.run(circuits=ReferenceCircuits.bell())
```
With this approach, we wouldn't even have to pass in `session` to `backend.run()` right? Something like this would work fine too

Reply:
Yes, this should work.